src/invocation/ClusterService.ts

```diff
@@ -14,317 +14,283 @@
+    public getMember(uuid: UUID): Member {
+        assertNotNull(uuid);
+        return this.memberListSnapshot.members.get(uuid.toString());
     }
 
     /**
-     * Connects to cluster. It uses the addresses provided in the configuration.
-     * @returns
+     * Gets the collection of members.
+     *
+     * @return The collection of members.
      */
-    connectToCluster(): Promise<void> {
-        return this.getPossibleMemberAddresses().then((res) => {
-            this.knownAddresses = [];
-            res.forEach((value) => {
-                this.knownAddresses = this.knownAddresses.concat(AddressHelper.getSocketAddresses(value));
-            });
-
-            const attemptLimit = this.client.getConfig().networkConfig.connectionAttemptLimit;
-            const attemptPeriod = this.client.getConfig().networkConfig.connectionAttemptPeriod;
-            return this.tryConnectingToAddresses(0, attemptLimit, attemptPeriod);
-        });
-    }
-
-    getPossibleMemberAddresses(): Promise<string[]> {
-        const addresses: Set<string> = new Set();
-
-        this.getMembers().forEach(function (member): void {
-            addresses.add(member.address.toString());
-        });
-
-        let providerAddresses: Set<string> = new Set();
-        const promises: Array<Promise<void>> = [];
-        this.client.getConnectionManager().addressProviders.forEach((addressProvider) => {
-            promises.push(addressProvider.loadAddresses().then((res) => {
-                providerAddresses = new Set([...Array.from(providerAddresses), ...res]);
-            }).catch((err) => {
-                this.logger.warn('Error from AddressProvider: ' + addressProvider, err);
-            }));
-        });
-        return Promise.all(promises).then(() => {
-            return Array.from(new Set([...Array.from(addresses), ...Array.from(providerAddresses)]));
-        });
+    public getMemberList(): Member[] {
+        return Array.from(this.memberListSnapshot.members.values());
     }
 
     /**
-     * Returns the list of members in the cluster.
-     * @returns
+     * Returns a collection of the members that satisfy the given {@link MemberSelector}.
+     *
+     * @param selector {@link MemberSelector} instance to filter members to return
+     * @return members that satisfy the given {@link MemberSelector}.
      */
-    getMembers(selector?: MemberSelector): Member[] {
-        if (selector === undefined) {
-            return this.members;
-        } else {
-            const members: Member[] = [];
-            this.members.forEach(function (member): void {
-                if (selector.select(member)) {
-                    members.push(member);
-                }
-            });
-            return members;
-        }
-    }
-
-    getMember(uuid: string): Member {
-        for (const member of this.members) {
-            if (member.uuid === uuid) {
-                return member;
+    public getMembers(selector: MemberSelector): Member[] {
```

Comment:
The `selector` argument has to be optional, as defined in the `Cluster` interface. This also means that the `undefined` case should be handled here.

Reply:
I removed the public `getMemberList` and handled the undefined or null case here by returning the non-filtered member list